File: backend/app/schemas/staff_profile.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class StaffProfileRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    person_id: int
    first_name: str
    last_name: str
    email: str | None
    employee_code: str
    is_active: bool
    is_available: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/staff_profile.ts
export interface StaffProfile {
  id: number;
  person_id: number;
  first_name: string;
  last_name: string;
  email?: string | null;
  employee_code: string;
  is_active: boolean;
  is_available: boolean;
  created_at: string;
  updated_at: string;
}
